### PR TITLE
Feature/pf 57-add-pet-front-end

### PR DESCRIPTION
**Pull request by agisulyok**

### CODE REVIEW

File: src/components/AddPetForm.jsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                            <input type="date" id="nextCheckUp" name="nextCheckUp" value={formData.nextCheckUp} onChange={handleChange}/>
+                        </div>
+                        <div>
+                            <label htmlFor="specialCondition">Special Condition:</label>
+                            <textarea id="specialCondition" name="specialCondition" value={formData.specialCondition} onChange={handleChange} rows="4" cols="50" placeholder="Enter any special conditions or notes about the pet..." />
+                        </div>
+                        <button type="submit" className="formButton">Add</button>
+                    </form>
+                </div>
+                <p style={{ color: 'red' }}>{error ? error : ""}</p>
```
Comment (markkovari) [inline src/components/AddPetForm.jsx:95]: I think you should not render the `<p>` tag only in the case if the error is not null, nut just its inner content. YOu can achieve it with the following:
```suggestion
                {error && <p style={{ color: 'red' }}>{error}</p>}
```

File: tests/AddPetForm.test.jsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        "petBreed": "dog",
+        "petSex": "Male",
+        "petBirthDate": "2024-01-01",
+        "lastCheckUp": "2024-06-15",
+        "nextCheckUp": "2024-08-15",
+        "specialCondition": "No conditions"
+    },
+      { headers: { 
+        "Content-Type": "application/json",
+        "Authorization": "Bearer null"
```
Comment (markkovari) [inline tests/AddPetForm.test.jsx:103]: This is looking relatively good, but you might run into an error that this thing is alway `null`. Maybe you wanted to add the token option to it too? If I understand this part should only work if you provide a token.

But if you have a test with `null`, you might have a false positive test.

**Review by markkovari — COMMENTED** [review]
Looking good so far, just a few nits. Please keep in mind that I added some comments on the backend side as well regarding the path.

File: tests/AddPetForm.test.jsx (hunk 2 above)
Comment (agisulyok) [reply tests/AddPetForm.test.jsx:103]: Hi Mark, I tried to mock the token, but somehow it always has Bearer null in the mockpost request. I tried like this:
const mockToken = 'your-mocked-token';
vi.spyOn(localStorage, 'getItem').mockImplementation((key) => {
  if (key === 'token') {
    return mockToken;
  }
  return null;
});

Do you might have any suggestion how I whould do this correctly?

**Review by Vozard — APPROVED** [review]
Can be merged...

**Review by AIRewan — APPROVED** [review]

**Pull request merged**